### PR TITLE
feat: add schema revision samples

### PR DESCRIPTION
**Pull request by feywind**
This adds support for easily getting a schema gapic client, along with samples for schema revisions.

### CODE REVIEW

**Comment by feywind** [comment]
Hmm. The sample tests are going to need some more manual testing.

**Review by hongalex — APPROVED** [review]
These samples look good. I think there are a few missing ones,most important is probably `pubsub_get_schema_revision` but also `pubsub_commit_proto_schema` and `pubsub_subscribe_avro_records_with_revisions`. The first two should probably go in this PR, while the latter two can be in a separate PR?

File: samples/README.md [hunk 1]
```diff
@@ -356,6 +400,25 @@ __Usage:__
 
 
 
+### Delete a Schema Revision
+
+Deletes a new schema revision on a project
```
Comment (kamalaboulhosn) [inline samples/README.md:405]: Remove "new"

File: samples/deleteSchemaRevision.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * This application demonstrates how to perform basic operations on
+ * schemas with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+// sample-metadata:
+//   title: Delete a Schema Revision
+//   description: Deletes a new schema revision on a project
```
Comment (kamalaboulhosn) [inline samples/deleteSchemaRevision.js:29]: Remove "new"

File: samples/typescript/deleteSchemaRevision.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * This application demonstrates how to perform basic operations on
+ * schemas with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+// sample-metadata:
+//   title: Delete a Schema Revision
+//   description: Deletes a new schema revision on a project
```
Comment (kamalaboulhosn) [inline samples/typescript/deleteSchemaRevision.ts:25]: Remove "new"

**Review by kamalaboulhosn — APPROVED** [review]

**Comment by feywind** [comment]
> These samples look good. I think there are a few missing ones,most important is probably `pubsub_get_schema_revision` but also `pubsub_commit_proto_schema` and `pubsub_subscribe_avro_records_with_revisions`. The first two should probably go in this PR, while the latter two can be in a separate PR?

Hmm, those didn't make it into my hotlist. I'll just add them in this PR.

**Review by hongalex — APPROVED** [review]

**Pull request merged**